hack/lib/golang.sh

```diff
@@ -393,7 +401,7 @@ kube::golang::place_bins() {
     # The substitution on platform_src below will replace all slashes with
     # underscores.  It'll transform darwin/amd64 -> darwin_amd64.
     local platform_src="/${platform//\//_}"
-    if [[ $platform == $host_platform ]]; then
+    if [[ $platform == "$host_platform" ]]; then
```

Comment:
why quote one but not both?

Reply:
That was pattern matching... The right side of the `==` operator is considered a pattern and is matched, and the quoting just forces it to be matched as a string.

Consider this script:

    #!/bin/bash
  
    bar="foobar"
    foo="foo*"
    
    [[ $bar == $foo ]] && echo "first true"
    [[ "$bar" == "$foo" ]] && echo "second true"
    [[ $bar == "$foo" ]] && echo "third true"
    [[ "$bar" == $foo ]] && echo "fourth true"

I get the output:

    first true
    fourth true

I think the normal use is to typically just quote the right hand side, but quoting the left hand side will not hurt. I'll do that and also improve the commit msg with the example.